frontend/src/ui/time.tsx

```diff
@@ -32,21 +37,27 @@ export const RelativeDate: React.FC<RelativeDateProps> = ({ date }) => {
 
     const prettyDate = (() => {
         const intl = new Intl.RelativeTimeFormat(i18n.language);
+        let timeAndUnit: [number, Intl.RelativeTimeFormatUnit];
         if (secsDiff <= 55) {
-            return intl.format(-secsAgo, "second");
+            timeAndUnit = [secsAgo, "second"];
         } else if (secsDiff <= 55 * MINUTE) {
-            return intl.format(-Math.round(secsAgo / MINUTE), "minute");
+            timeAndUnit = [secsAgo / MINUTE, "minute"];
         } else if (secsDiff <= 23 * HOUR) {
-            return intl.format(-Math.round(secsAgo / HOUR), "hour");
+            timeAndUnit = [secsAgo / HOUR, "hour"];
         } else if (secsDiff <= 6 * DAY) {
-            return intl.format(-Math.round(secsAgo / DAY), "day");
+            timeAndUnit = [secsAgo / DAY, "day"];
         } else if (secsDiff <= 3.5 * WEEK) {
-            return intl.format(-Math.round(secsAgo / WEEK), "week");
+            timeAndUnit = [secsAgo / WEEK, "week"];
         } else if (secsDiff <= 11 * MONTH) {
-            return intl.format(-Math.round(secsAgo / MONTH), "month");
+            timeAndUnit = [secsAgo / MONTH, "month"];
         } else {
-            return intl.format(-Math.round(secsAgo / YEAR), "year");
+            timeAndUnit = [secsAgo / YEAR, "year"];
         }
+        const [time, unit] = timeAndUnit;
+        const prefix = time < 0 ? "video.starts-in" : "video.started";
+        return isLive
+            ? t(prefix, { duration: intl.format(Math.round(-time), unit) })
+            : intl.format(Math.round(-time), unit);
```

Comment:
Deduplicating some logic.

```suggestion
        const relative = intl.format(Math.round(-time), unit);
        return isLive ? t(prefix, { duration: relative }) : relative;
```